report/map_keys.go

```diff
@@ -0,0 +1,192 @@
+	DockerRestartContainer:       DockerRestartContainer,
+	DockerPauseContainer:         DockerPauseContainer,
+	DockerUnpauseContainer:       DockerUnpauseContainer,
+	DockerRemoveContainer:        DockerRemoveContainer,
+	DockerAttachContainer:        DockerAttachContainer,
+	DockerExecContainer:          DockerExecContainer,
+	DockerContainerName:          DockerContainerName,
+	DockerContainerCommand:       DockerContainerCommand,
+	DockerContainerPorts:         DockerContainerPorts,
+	DockerContainerCreated:       DockerContainerCreated,
+	DockerContainerNetworks:      DockerContainerNetworks,
+	DockerContainerIPs:           DockerContainerIPs,
+	DockerContainerHostname:      DockerContainerHostname,
+	DockerContainerIPsWithScopes: DockerContainerIPsWithScopes,
+	DockerContainerState:         DockerContainerState,
+	DockerContainerStateHuman:    DockerContainerStateHuman,
+	DockerContainerUptime:        DockerContainerUptime,
+	DockerContainerRestartCount:  DockerContainerRestartCount,
+	DockerContainerNetworkMode:   DockerContainerNetworkMode,
+
+	KubernetesName:                 KubernetesName,
+	KubernetesNamespace:            KubernetesNamespace,
+	KubernetesCreated:              KubernetesCreated,
+	KubernetesIP:                   KubernetesIP,
+	KubernetesObservedGeneration:   KubernetesObservedGeneration,
+	KubernetesReplicas:             KubernetesReplicas,
+	KubernetesDesiredReplicas:      KubernetesDesiredReplicas,
+	KubernetesNodeType:             KubernetesNodeType,
+	KubernetesGetLogs:              KubernetesGetLogs,
+	KubernetesDeletePod:            KubernetesDeletePod,
+	KubernetesScaleUp:              KubernetesScaleUp,
+	KubernetesScaleDown:            KubernetesScaleDown,
+	KubernetesUpdatedReplicas:      KubernetesUpdatedReplicas,
+	KubernetesAvailableReplicas:    KubernetesAvailableReplicas,
+	KubernetesUnavailableReplicas:  KubernetesUnavailableReplicas,
+	KubernetesStrategy:             KubernetesStrategy,
+	KubernetesFullyLabeledReplicas: KubernetesFullyLabeledReplicas,
+	KubernetesState:                KubernetesState,
+	KubernetesIsInHostNetwork:      KubernetesIsInHostNetwork,
+	KubernetesRestartCount:         KubernetesRestartCount,
+	KubernetesMisscheduledReplicas: KubernetesMisscheduledReplicas,
+	KubernetesPublicIP:             KubernetesPublicIP,
+	KubernetesSchedule:             KubernetesSchedule,
+	KubernetesSuspended:            KubernetesSuspended,
+	KubernetesLastScheduled:        KubernetesLastScheduled,
+	KubernetesActiveJobs:           KubernetesActiveJobs,
+
+	ECSCluster:             ECSCluster,
+	ECSCreatedAt:           ECSCreatedAt,
+	ECSTaskFamily:          ECSTaskFamily,
+	ECSServiceDesiredCount: ECSServiceDesiredCount,
+	ECSServiceRunningCount: ECSServiceRunningCount,
+	ECSScaleUp:             ECSScaleUp,
+	ECSScaleDown:           ECSScaleDown,
+}
+
+// lookupCommonKey tris to avoid an allocation in populating the key
+// by looking it up
+func lookupCommonKey(key *string, b []byte) {
```

Comment:
I would just have this return the string rather than taking a pointer.

Reply:
I am pretty sure I tried that and got more allocations.

Reply:
Just tried again and it's fine. Will change.